modules/haloRtdProvider.js

```diff
@@ -11,27 +11,77 @@ import {submodule} from '../src/hook.js';
 import {ajax} from '../src/ajax.js';
 import { getStorageManager } from '../src/storageManager.js';
 
-const storage = getStorageManager();
+export const storage = getStorageManager();
 
 /** @type {string} */
 const MODULE_NAME = 'realTimeData';
-const SUBMODULE_NAME = 'audigent';
-const HALOID_LOCAL_NAME = 'auHaloId';
-const SEG_LOCAL_NAME = '__adgntseg';
+const SUBMODULE_NAME = 'halo';
+const ERR_MSG = 'AUDIGENT_ERR';
+
+export const HALOID_LOCAL_NAME = 'auHaloId';
+export const SEG_LOCAL_NAME = '__adgntseg';
+
+const set = (obj, path, val) => {
+  const keys = path.split('.');
+  const lastKey = keys.pop();
+  const lastObj = keys.reduce((obj, key) => obj[key] = obj[key] || {}, obj);
+  lastObj[lastKey] = lastObj[lastKey] || val;
+};
+
+/** bid adapter format segment augmentation functions */
+const segmentMappers = {
+  appnexus: function(bid, segments) {
+    set(bid, 'params.user.segments', []);
+    let appnexusSegments = [];
+    segments.forEach(segment => {
+      if (typeof segment.value != 'undefined' && segment.value != null) {
+        let appnexusSegment = {'id': segment.id, 'value': segment.value};
+        appnexusSegments.push(appnexusSegment);
+      }
+    })
+    bid.params.user.segments = bid.params.user.segments.concat(appnexusSegments);
+  },
+  generic: function(bid, segments) {
+    bid.segments = segments;
```

Comment:
Wondering if in this case we want to also set the bid with preexisting segments from the bid and concat to the bid in the same way as appnexus? IAB docs show segments within the user object I imagine this could be as simple as using the same line from 34 and mentioning this in the markdown for the generic approach.

Reply:
this 'generic' case is mostly for testing.. we will probably expand it to align with prebid's unified taxonomy generic segments later on.  until then, we are adding fpd segment data that can be consumed by individual bidders on line 66